### DISCUSSION TITLE
Why does my SNZB-02D report only even digits?

### DISCUSSION THREAD

**Discussion by Ldoppea** [Koenkk/zigbee2mqtt#21134] [Devices]
Hi,

I'm using a [SNZB-02D](https://www.zigbee2mqtt.io/devices/SNZB-02D.html#sonoff-snzb-02d) with Z2M and I'm struggling configuring it.

Both its screen and Z2M only display temperatures with even digits. For example I can get `20.2°C`, `20.4°C` or `20.6°C`. But I never get `20.3°C`, `20.5°C` nor `20.7°C`

In Z2M I tried to reduce the `min rep change` value to `10` or the `Temperature precision` to `3` but nothing changes.

I was going to conclude that this is a limitation on the device, but then I noticed that both the official Sonoff pictures and the device's package display uneven digits
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1884255/11d68d3c-f17d-470b-bf08-8193b3ac565b)

Also after some research I found a youtube review showing the device's temp rising from `12.5°C` to `12.8°C` which seems to prove that my device's behavior is not normal (I since found some other reviews confirming that the devices can display both even and uneven numbers)

Am I missing some configuration? Can someone help me understanding what is happening?

Thanks

**Comment by Dominik-Sikora** [comment]
I tested it last week, and unfortunately, the sensor reported changes when they were at least 0.2°C or 1.0%. If you warm it up rapidly, you can see uneven digits. It can go up by 0.3°C and later slowly down by 0.2°C. The basic reporting functionality of the sensor is done independently of the Z2M reporting settings. Such strange behavior disqualifies that sensor for me. Maybe Sonoff will upgrade the firmware. I'm looking for another sensor with a display. I have 10 pieces of SNZB-02 sensors (without display), and they operate without reporting issues. The changes are reported even if the temperature or humidity varies by 0.01.

**Reply by Ldoppea** [reply]
Hi,

Sorry I missed the notification about your reply.

Thanks for your answer

> If you warm it up rapidly, you can see uneven digits. It can go up by 0.3°C and later slowly down by 0.2°C

Well... that may explain why some reviews show them with uneven digits. I suppose testers did not want to wait hours for their tests and so they warmed it up fast enough to make this happen

**Reply by cbrunnkvist** [reply]
there is a "Min Reporting" setting for the device instance in z2m: could it be that "don't push zigbee event unless delta is > 0.2" is adding to the confusion?
